refactor: restructure test framework and update entry points

danjoa opened the pull request:
This PR brings the package in better shape (foreseen long time). 
Noteworthy: 

- Removes all hard dependencies to `@sap/cds` → only required for `cds.test.data`
- Replaces UNIX `find` by programmatic `fetch` function → en par with jest's but faster 
- Adds test fixture for `chest` so `describe` et al are available globally
- Adds `jest` global with mock support so most of our internal tests also run with `chest`

chgeo commented:
> Removes all hard dependencies to `@sap/cds` → only required for `cds.test.data`

I see that it's not needed for the test runner, but don't we need it for any `cds.test()` calls because we start the CAP server then?
We could go for an optional peerDependency here.  W/o it, npm might install `@sap/cds` at a place where we can't load it.

danjoa commented:
> I see that it's not needed for the test runner, but don't we need it for any `cds.test()` calls because we start the CAP server then?

Also pondered about this and came to that conclusion, which I wanted to discuss:
Tests today always use `cds.test` like that: 

```js
import cds from '@sap/cds'
const { GET, expect } = cds.test()
```

And if we stay with that, there would not be this de-route via `@sap/cds-test`

danjoa commented:
I wonder where [this error](https://github.com/cap-js/cds-test/actions/runs/13894903752/job/38873291606?pr=18) comes from? With this PR, `chest` doesn't use UNIX find anymore → looks as if the former implementation is somehow still alive?

BobdenOs commented:
> I wonder where [this error](https://github.com/cap-js/cds-test/actions/runs/13894903752/job/38873291606?pr=18) comes from? With this PR, `chest` doesn't use UNIX find anymore → looks as if the former implementation is somehow still alive?

When I checkout this branch and run `npm ci` locally the `node_modules/.bin/chest` file is the one provided by `@cap/cds`. When I run `npm i @cap-js/cds-test@.` the file is updated with the `@cap-js/cds-test` version of the file. This seems to be undefined behavior in `npm` when two pacakges provide the same commands.

danjoa commented:
Thanks @BobdenOs → fixed it in `@sap/cds`

chgeo commented:
> > I see that it's not needed for the test runner, but don't we need it for any `cds.test()` calls because we start the CAP server then?
> 
> Also pondered about this and came to that conclusion, which I wanted to discuss: Tests today always use `cds.test` like that:
> 
> ```js
> import cds from '@sap/cds'
> const { GET, expect } = cds.test()
> ```
> 
> And if we stay with that, there would not be this de-route via `@sap/cds-test`

I think we should stay with that pattern, otherwise many tests in apps need to be adjusted.
And it's in line with the rest of our API that starts in the `@sap/cds` facade, no matter if the implementation is in a different (library) package. See the `cds.compile.to...` functions that are located in different plugins).

chgeo commented:
> Fully agreed. Which means [this concern](https://github.com/cap-js/cds-test/pull/18#issuecomment-2728601679) is resolved, and we can indeed remove the dependency to `@sap/cds` as in this PR, correct?

True, such a dependency seems not needed from the viewpoint of the application as it has a dependency to `@sap/cds` anyway.

However, from the viewpoint of the `cds-test` impl., which does `require('@sap/cds')`, there should be some sort of dependency backing this call.  And this is what [peerDependencies](https://docs.npmjs.com/cli/v11/configuring-npm/package-json#peerdependencies) are for.
We have this pattern in all/most other CAP plugins as well.

So, in practice the peerDependency doesn't make a difference, but is good style I'd say.

danjoa commented:
- Restored the peer dependency to `@sap/cds`.
- Replaced all these absolute package refs by relative ones (as usual, isn't it?).
- Removed those self-referencing dev dependency which were necessary because of that.  
- Added workarounds for former `@sap/cds` overriding our `chest` binary

@chgeo → could be merged now?

danjoa commented:
Ups... got merged automatically ... did you approve before?

chgeo commented:
> Ups... got merged automatically ... did you approve before?

No, not really.

danjoa commented:
So auto-merge seems to be a free run in here
